src/component/Signalment.js

```diff
@@ -1,59 +1,79 @@
 import React from 'react';
+import {connect} from 'react-redux';
 import style from 'antd/dist/antd.css';
 import {Form, Col,Select, notification, Input, Switch, Button,Icon } from 'antd';
 import {BrowserRouter,Route,Link} from 'react-router-dom';
+import {store} from '../store/store';
+import {usersignalmentDetails} from '../action/action'
 
+class SignalmentForm extends React.Component {
 
-class SingnalmentForm extends React.Component {
-
-  onChange = (checked)=> {
-    console.log(`switch to ${checked}`);
+  constructor(props) {
+    super(props);
+    this.state = {
+      Distance:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].Distance,
```

Comment:
y array ?
props.signalmentDetails[props.signalmentDetails.length-1]
same comment is mentioned in saurabh and aditya

Reply:
sir signalmentDetails is an array as all the details stored are in form of an object